build.sbt

```diff
@@ -579,14 +621,48 @@ lazy val enableMacros = macroDependencies(hardcore = false)
 lazy val enableHardcoreMacros = macroDependencies(hardcore = true)
 
 def macroDependencies(hardcore: Boolean) = libraryDependencies ++= {
-  val scalaReflect = Seq("org.scala-lang" % "scala-reflect" % scalaVersion.value % "provided")
+  val scalaReflect =
+    Seq("org.scala-lang" % "scala-reflect" % scalaVersion.value % "provided")
   val scalaCompiler = {
-    if (hardcore) Seq("org.scala-lang" % "scala-compiler" % scalaVersion.value % "provided")
+    if (hardcore)
+      Seq("org.scala-lang" % "scala-compiler" % scalaVersion.value % "provided")
     else Nil
   }
   val backwardCompat210 = {
-    if (scalaVersion.value.startsWith("2.10")) Seq("org.scalamacros" %% "quasiquotes" % "2.1.0")
+    if (scalaVersion.value.startsWith("2.10"))
+      Seq("org.scalamacros" %% "quasiquotes" % "2.1.0")
     else Seq()
   }
   scalaReflect ++ scalaCompiler ++ backwardCompat210
 }
+
+lazy val compilePublish: Def.Initialize[Task[Unit]] = Def.taskDyn {
```

Comment:
The reason I refactored the build so much is because the crossProject changes made some settings evaluate twice or three times on build reload. While moving things around, I got weird compiler errors when around `Def.taskDyn` which I believe are attributed to being inside the `:=` macro. Even if this is only called once I think it's clearer to move it outside of the long `publishableSettings` sequence. 

I can move it back if you want. I honestly think there's way to much logic in this build.sbt which should be refactored into an sbt plugin, but that's for another PR.